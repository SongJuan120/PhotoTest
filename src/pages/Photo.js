import React, {useState, useEffect} from 'react'
import axios from 'axios'
import { connect } from 'react-redux';
import {getImageActions} from '../store/actions/getImageActions'
import '../assets/styles/style.css'

const Photo = (props) => {            
    const [ImgList, setImgList] = useState ([])  
    
    useEffect (()=> {
        getImages (1)
    }, [])

    const getImages = (page) => {
        props.getImageData (page)
    }   
   
    return (
        <main className='App'>            
            <div>
                Hello World
            </div>
        </main>
    )
}

const mapStateToProps = (state) => {
    return {
        items: state.items,
        hasErrored: state.itemsHasErrored,
        isLoading: state.itemsIsLoading
    };
};
const mapDispatchToProps = (dispatch) => {
    return {
        getImageData: (page) => dispatch(getImageActions(page))
    };
};
export default connect(mapStateToProps, mapDispatchToProps)(Photo);
