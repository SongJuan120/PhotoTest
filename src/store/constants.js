export const getImageConstants = {
  GET_IMAGES: 'GET_IMAGES',
  GET_IMAGES_FAILURE: 'GET_IMAGES_FAILURE',  
  GET_IMAGES_SUCCESS: 'GET_IMAGES_SUCCESS',    
};
