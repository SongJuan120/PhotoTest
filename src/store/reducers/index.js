import { combineReducers } from 'redux';
import imageReducer from './imageReducer';

const rootReducer = combineReducers({
  imageReducer,
});

export default rootReducer;