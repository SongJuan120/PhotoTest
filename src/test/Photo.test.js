import React from 'react';
import renderer from 'react-test-renderer';
import Photo from "../pages/Photo";
import { store } from '../store/store';
import { connect, Provider } from 'react-redux';


test('render Photo component', () => {
  const minProps = {
      isLoading: false,
      data: [],
      err: null
  };
  const component = renderer.create(
    <Provider store={store}>
      <Photo {...minProps} connect />
    </Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

