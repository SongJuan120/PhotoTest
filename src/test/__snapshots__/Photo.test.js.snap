// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Photo component 1`] = `
<main
  className="App"
>
  <div
    className="container"
    onScroll={[Function]}
  >
    <div
      className="section"
    />
    <div
      className="section"
    />
    <div
      className="section"
    />
  </div>
</main>
`;
